src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -898,6 +878,138 @@ public static int Sum<T>(this ImmutableArray<T> items, Func<T, int> selector)
             return dictionary;
         }
 
+        internal static void AddToMultiValueDictionaryBuilder<K, T>(Dictionary<K, object> accumulator, K key, T item)
+            where K : notnull
+            where T : notnull
+        {
+            if (accumulator.TryGetValue(key, out var existingValueOrArray))
+            {
+                if (existingValueOrArray is ArrayBuilder<T> arrayBuilder)
+                {
+                    // Already a builder in the accumulator, just add to that.
+                }
+                else
+                {
+                    // Just a single value in the accumulator so far.  Convert to using a builder.
+                    arrayBuilder = ArrayBuilder<T>.GetInstance(capacity: 2);
+                    arrayBuilder.Add((T)existingValueOrArray);
+                    accumulator[key] = arrayBuilder;
+                }
+
+                arrayBuilder.Add(item);
+            }
+            else
+            {
+                // Nothing in the dictionary so far.  Add the item directly.
+                accumulator.Add(key, item);
+            }
+        }
+
+        internal static void CreateNameToMembersMap
+            <TNamespaceOrTypeSymbol, TNamedTypeSymbol, TNamespaceSymbol>
+            (Dictionary<string, object> dictionary, Dictionary<String, ImmutableArray<TNamespaceOrTypeSymbol>> result)
+            where TNamespaceOrTypeSymbol : class
+            where TNamedTypeSymbol : class, TNamespaceOrTypeSymbol
+            where TNamespaceSymbol : class, TNamespaceOrTypeSymbol
+        {
+            foreach (var kvp in dictionary)
+            {
+                object value = kvp.Value;
+                ImmutableArray<TNamespaceOrTypeSymbol> members;
+
+                var builder = value as ArrayBuilder<TNamespaceOrTypeSymbol>;
+                if (builder != null)
+                {
+                    Debug.Assert(builder.Count > 1);
+                    bool hasNamespaces = false;
+                    for (int i = 0; (i < builder.Count) && !hasNamespaces; i++)
```

Comment:
> && !hasNamespaces [](http://example.com/codeflow?start=55&length=18)

Is this check needed?

Reply:
> Is this check needed?

This function is an existing code moved from C# compiler. It is not my goal to change what it is doing

Reply:
It looks like the body of the `for` loop changed from `NameToSymbolMapBuilder.CreateMap()` so the check here is no longer needed.

Reply:
> It looks like the body of the for loop changed from ```NameToSymbolMapBuilder.CreateMap()``` so the check here is no longer needed.

That is fair